pkg/util/pullsecret/pullsecret.go

```diff
@@ -89,6 +110,32 @@ func Merge(_base, _ps string) (string, bool, error) {
 	return string(b), changed, err
 }
 
+// FixPullSecretData checks if the userData Secret Auth keys are not equal to the operator Auth keys
+// if they are not they are considered broken and are replaced by the ones stored in operator Data
+// if key is not found in operatorData it is not taken care of as this is user problem
+// Function is idempotent
+func FixPullSecretData(operatorData, userData *SerializedAuthMap) (fixedSecret *SerializedAuthMap, fixed bool) {
```

Comment:
We have an issue here. 
Lets say customer updated their `redhat.io` secret in the cluster with the new value (People changed roles, cluster moved to new team, etc). So they replaced the secret with new one. We still have old key in our cache. We should not care about anything else apart ACR secret and we should fix only ours secret.  In this case if we have stale customer redhat.io secret we will be overriding it for the customer.

Reply:
I am moving this one to separate PR. It will make the review easier.